core/src/main/java/org/apache/calcite/prepare/Prepare.java

```diff
@@ -143,10 +143,18 @@ protected RelRoot optimize(RelRoot root,
 
     final List<RelOptMaterialization> materializationList = new ArrayList<>();
     for (Materialization materialization : materializations) {
+      RelOptTable table = materialization.tableRel.getTable();
+      List<String> qualifiedTableName;
+      if (table == null) {
+        qualifiedTableName = materialization.materializedTable.path();
```

Comment:
Just to confirm, can {{qualifiedTableName = materialization.materializedTable.path();}} work for both situations or this if-else block is inevitable?

Reply:
Seems it should work, I'll test that and push the change if it works.